setup.py

```diff
@@ -26,7 +26,16 @@ def read_file(filename):
     package_data={},
     python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*",
     setup_requires=["setuptools_scm"],
-    install_requires=["click>=6", "six"],
+    install_requires=[
+        # direct dependencies
+        "click>=6",
+        "pip",
+        "six",
+        # indirect dependencies
+        "setuptools",  # typically needed when invoking setup.py
```

Comment:
@codingjoe , this is *not* about the running of the pip-tools `setup.py` -- that is what `setup_requires` is for, and `setuptools` is implied there, and I agree it is not needed unless a specific minimum setuptools version is needed.  Since pip-tools offers wheels on PyPI, it is even possible to install pip-tools without setuptools being pre-installed at all.  I regularly set up minimal Docker images, and I try to avoid having setuptools installed if possible, and increasingly that is possible if the dependencies of packages are properly declared.

This PR is about the run-time dependencies of pip-tools *after* it is installed.  pip-tools invokes the `setup.py` of other packages, and setuptools is the most basic dependency for that.

It is *not* necessarily implied that setuptools is installed after pip-tools is installed.  It needs to be listed in `install_requires` to declare that `setuptools` is needed after installation has been completed.  This metadata is exposed in wheels, and that metadata is used by all sorts of tools.

Reply:
Good point, it does make sense to have it as a runtime dependency listed here.